apps/web/src/components/templates/TemplateFormProvider.tsx

```diff
@@ -1,42 +1,76 @@
 import { FormProvider, useFieldArray, useForm } from 'react-hook-form';
 import { FieldArrayProvider } from './FieldArrayProvider';
 import { IForm } from './use-template-controller.hook';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
 import { ChannelTypeEnum } from '@novu/shared';
 
-export const TemplateFormProvider = ({ children }) => {
-  const methods = useForm<IForm>({
-    resolver: async (data) => {
-      const errors: any = {};
-      let values = data;
-      if (!data.name) {
-        errors.name = 'Required field name';
-      }
-
-      if (!data.notificationGroup) {
-        errors.notificationGroup = 'Required field notification group';
-      }
-
-      data.steps.forEach((step, index) => {
-        if (
-          (step.template.type === ChannelTypeEnum.SMS || step.template.type === ChannelTypeEnum.IN_APP) &&
-          step.template.content.length === 0
-        ) {
-          errors[`steps.${index}.template.content`] = 'Required field content';
+const schema = z
+  .object({
+    name: z.string({
+      required_error: 'Required field name',
```

Comment:
I think we should do a better error message as it sometimes shown not adjacent to the actual input. 
What do you think about
'Required - Notification Name' ,  
'Required - Notification Group' , 
 'Required - Message Content'  (for in app and SMS) , 
'Required - Email Subject'

Reply:
Yes lets discuss those in a Monday meeting 😄